Assets/Scripts/GameObjectsScripts/Food.cs

```diff
@@ -6,7 +6,7 @@ public class Food : MonoBehaviour {
 
     private void OnTriggerEnter2D(Collider2D collision)
     {
-		if (collision.gameObject.GetComponent<Interactive>() != null)
+		if (collision.gameObject.GetComponent<Interactive>() != null && collision.gameObject.tag == "Player")
```

Comment:
This change will prevent food from interacting with other types of collisions ( consider if we want to write a hook that can hook food for us in the future ). I think the food being destroyed by projectiles can be considered either as a feature, or if you don't like it, a bug. You can try to "fix" it in some other way if you don't like it, otherwise just consider it as a feature because it also adds a risk to firing a projectile.